include/irr/asset/filters/CSwizzleAndConvertImageFilter.h

```diff
@@ -188,20 +188,50 @@ class CSwizzleAndConvertImageFilter<EF_UNKNOWN,EF_UNKNOWN,Swizzle> : public CIma
 			#endif
 			auto perOutputRegion = [&blockDims,inFormat,outFormat,&state](const CMatchedSizeInOutImageFilterCommon::CommonExecuteData& commonExecuteData, CBasicImageFilterCommon::clip_region_functor_t& clip) -> bool
 			{
-				auto swizzle = [&commonExecuteData,&blockDims,inFormat,outFormat,&state](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos)
+				const auto inTexelOrBlockByteSize = asset::getTexelOrBlockBytesize(inFormat);
+				const auto inChannelAmount = asset::getFormatChannelCount(inFormat);
+
+				// TESTING
+				auto getClampedSourcePixels = [&](auto sourcePixels)
+				{
+					constexpr auto MaxPlanes = 4;
+					auto buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(blockDims.y * blockDims.x * inTexelOrBlockByteSize);
```

Comment:
you just want a `double pixelVals[4];` instead of the buffer.

`decodePixels` takes an intrablock X,Y coordinate  to decode one pixel at a time.

Reply:
adjusted in following commit